src/block.rs

```diff
@@ -1,60 +1,68 @@
-//! Masking reqions of a sequence
+use std::ops::{Bound, Deref, Range, RangeBounds};
 
-use std::cmp::{max, min};
-use std::ops::Range;
-
-use crate::types::Field;
-
-/// A block mask for sequence regions
+/// Block mask for sequence regions
 ///
 /// Blocks are used to mark regions as soft-masked or hard-masked.
-#[derive(PartialEq, Debug, Clone, Copy)]
-pub struct Block {
-    pub start: Field,
-    pub length: Field,
-}
+pub type Block = Range<usize>;
 
-impl Block {
-    /// Create a new block
-    pub fn new(start: Field, length: Field) -> Block {
-        Block { start, length }
-    }
+/// Sorted collection of block masks
+#[derive(Debug, Clone)]
+pub struct Blocks(pub Vec<Block>);
 
-    /// Determine an overlap with another block
-    ///
-    /// Convenience function to determine overlaps between sequence regions.
-    ///
-    /// Note: Blocks of the same type should usually not overlap in practise.
-    pub fn overlap(&self, other: &Block) -> Option<Range<usize>> {
-        let start = max(self.start, other.start);
-        let end = min(self.start + self.length, other.start + other.length);
-        if start < end {
-            Some(start as usize..end as usize)
-        } else {
-            None
-        }
+impl Deref for Blocks {
+    type Target = [Block];
+
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        &self.0
     }
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
+impl Blocks {
+    #[inline]
+    pub fn iter_overlaps(&self, range: impl RangeBounds<usize>) -> impl Iterator<Item = &Block> {
```

Comment:
I personally would call the function just `overlaps`. I don't see any functions in the stdlib that are called `iter_something`. But I would also be fine with keeping it as it currently is.

Reply:
Yep, agreed. In stdlib it's all like `keys()`, `values()`, etc, so `overlaps()` is more rustic.